cpp/include/cudf/detail/utilities/rmm_host_vector.hpp

```diff
@@ -109,30 +109,6 @@ class rmm_host_allocator {
   {
   }
 
-  /**
-   * @brief Copy constructor
-   */
-  rmm_host_allocator(rmm_host_allocator const& other) = default;
-
-  /**
-   * @brief Move constructor
-   */
-  rmm_host_allocator(rmm_host_allocator&& other) = default;
-
-  /**
-   * @brief Assignment operator
-   */
-  rmm_host_allocator& operator=(rmm_host_allocator const& other)
-  {
-    mr = other.mr;
```

Comment:
I notice that the equality operator `==` also only checks mr but not stream. Seems it's intentional but not sure why. Worth double checking with @nvdbaranec 

Could we set up a stream mode test for `cpp/tests/utilities_tests/io_utilities_tests.cpp`? In this way, we can avoid/uncover similar issues before delivery.

Reply:
added stream comparison to the operator==